pkg/platform/runtime/buildscript/buildscript.go

```diff
@@ -153,3 +174,26 @@ func (f *FuncCall) String() string {
 	buf.WriteString(")")
 	return buf.String()
 }
+
+func (s *Script) EqualsBuildExpressionBytes(exprBytes []byte) bool {
+	expr, err := buildexpression.New(exprBytes)
+	if err != nil {
+		multilog.Error("Unable to create buildexpression from incoming JSON: %v", err)
+		return false
+	}
+	return s.EqualsBuildExpression(expr)
+}
+
+func (s *Script) EqualsBuildExpression(expr *buildexpression.BuildExpression) bool {
+	myJson, err := json.Marshal(s.Expr)
+	if err != nil {
+		multilog.Error("Unable to marshal this buildscript to JSON: %v", err)
+		return false
+	}
+	otherJson, err := json.Marshal(expr)
+	if err != nil {
+		multilog.Error("Unable to marshal other buildscript to JSON: %v", err)
+		return false
+	}
+	return string(myJson) == string(otherJson)
+}
```

Comment:
It appears that these functions are unused.

Reply:
In this PR yes, but in the code base they are used in a few places.